modules/mesh-layers/src/scenegraph-layer/scenegraph-layer-vertex.glsl.js

```diff
@@ -1,36 +1,98 @@
 export default `\
+#if (__VERSION__ < 300)
+  #define _attribute attribute
+  #define _varying varying
+#else
+  #define _attribute in
+#define _varying out
+#endif
+
 // Instance attributes
-attribute vec3 instancePositions;
-attribute vec2 instancePositions64xy;
-attribute vec4 instanceColors;
-attribute vec3 instancePickingColors;
-attribute mat3 instanceModelMatrix;
-attribute vec3 instanceTranslation;
+_attribute vec3 instancePositions;
+_attribute vec2 instancePositions64xy;
+_attribute vec4 instanceColors;
+_attribute vec3 instancePickingColors;
+_attribute mat3 instanceModelMatrix;
+_attribute vec3 instanceTranslation;
 
 // Uniforms
 uniform float sizeScale;
 uniform mat4 sceneModelMatrix;
 
 // Attributes
-attribute vec4 POSITION;
+_attribute vec4 POSITION;
 
 #ifdef HAS_UV
-  attribute vec2 TEXCOORD_0;
-  varying vec2 vTEXCOORD_0;
+  _attribute vec2 TEXCOORD_0;
 #endif
-varying vec4 vColor;
 
-void main(void) {
+#ifdef MODULE_PBR
+  #ifdef HAS_NORMALS
+    _attribute vec4 NORMAL;
+  #endif
+
+  #ifdef HAS_TANGENTS
+    _attribute vec4 TANGENT;
+  #endif
+#endif
+
+// Varying
+// MODULE_PBR contains all the varying definitions needed
+#ifndef MODULE_PBR
+  _varying vec4 vColor;
+
   #ifdef HAS_UV
+    _varying vec2 vTEXCOORD_0;
+  #endif
+#endif
+
+// Utils
+#if defined(HAS_NORMALS) && defined(MODULE_PBR)
+  vec3 getProjectedNormal() {
+    return project_normal(instanceModelMatrix * (sceneModelMatrix * NORMAL).xyz);
```

Comment:
You do not need `project_normal` if normals are provided externally. This function projects lngLat delta to the common space in `LNG_LAT` mode